pkg/query-service/app/parser.go

```diff
@@ -1046,6 +1045,25 @@ func ParseQueryRangeParams(r *http.Request) (*v3.QueryRangeParamsV3, *model.ApiE
 					}
 				}
 			}
+
+			var timeShiftBy int64
+			if len(query.Functions) > 0 {
+				for idx := range query.Functions {
+					function := &query.Functions[idx]
+					if function.Name == v3.FunctionNameTimeShift {
+						// move the function to the beginning of the list
```

Comment:
why can't a user take the moving average of a time-shifted series?

Reply:
They can do it, it's just that the position of timeshift doesn't matter. Whether the data manipulation function comes after or before the timeshift produces the same value.

Reply:
and timeshift must be the 1st function to act in code as it changes the end and start and hence the input for the other functions to work on